### ISSUE TITLE
User package is tested

### ISSUE THREAD

**Issue by kenfus** [i4Ds/Karabo-Pipeline#498]
The user package was once tested, but not anymore. We definitely need to add this back. The tests inside karabo can easily be run as follows:

`pytest --pyargs karabo.test`

I think to make it easier, it can be run after building and uploading the package, to test the user experience. However, it's annoying that some requirements to run the test are inside the dev enviroment.yaml and in the requirements.txt

**Comment by Lukas113** [comment]
I agree we should test the user-build. What makes sense is to do so after building the user-build, but before uploading it to anaconda.org. So install the user-build with `conda install` from local, then call the `karabo.util.testing.run_tests` function, and then upload the user-build if the tests succeeded. Make sure to not have the repository or the dev virtual-environment from where you do the user-build test. I think all of that should be done in `conda-build.yml`.

Regarding to your post: I don't think that we ever tested the user-package properly. I don't agree with your statement that some things are in the environment.yaml (all direct dependencies of a karabo-package) and others in the requirements.txt (dev dependencies) since the purpose of these files should be clear.
